website/server/models/task.js

```diff
@@ -25,6 +26,13 @@ export let TaskSchema = new Schema({
   type: {type: String, enum: tasksTypes, required: true, default: tasksTypes[0]},
   text: {type: String, required: true},
   notes: {type: String, default: ''},
+  shortName: {
+    type: String,
+    validate: [function validateShortName (val) {
+      return Boolean(this.userId) && !validator.isUUID(val) && val.match(SHORT_NAME_VALIDATION_REGEX);
+    },
+    'Task short names can only contain alphanumeric characters, underscores and dashses. They must be owned by a user.'],
```

Comment:
How about:

'Task short names can only contain alphanumeric characters, underscores, and dashes. They cannot be UUIDs. They can only be applied to tasks in a user's own task list.'